modules/programs/chromium.nix

```diff
@@ -4,25 +4,53 @@ with lib;
 
 let
 
+  supportedBrowsers = [
+    "chromium"
+    "google-chrome"
+    "google-chrome-beta"
+    "google-chrome-dev"
+    "brave"
+  ];
+
   browserModule = defaultPkg: name: visible:
     let
       browser = (builtins.parseDrvName defaultPkg.name).name;
       isProprietaryChrome = hasPrefix "Google Chrome" name;
     in {
       enable = mkOption {
         inherit visible;
+        type = types.bool;
         default = false;
         example = true;
         description = "Whether to enable ${name}.";
-        type = lib.types.bool;
       };
 
       package = mkOption {
         inherit visible;
-        type = types.package;
+        type = types.nullOr types.package;
```

Comment:
Why not keep it as `types.package` and do
```
cfg.package.override { commandLineArgs = … }
```
if `cfg.commandLineArgs != []`?

Reply:
my original thought process was that the override would only occur if the package was left as default, similar to how the sway module handles this, as it is non trivial to tell the interface that a package uses (i.e a user may write their own chromium derivation which does not accept the formal argument).

https://github.com/nix-community/home-manager/blob/039f786e609fdb3cfd9c5520ff3791750c3eaebf/modules/services/window-managers/i3-sway/sway.nix#L319-L324

null was introduced as a hacky workaround because I am not sure how to compare derivations, I am not sure if a direct comparison yields the correct result.

Reply:
I have fixed the redundant null type issue by documenting the effects of `commandLineArgs` when `package` is set in the `commandLineArgs` option, and applied your suggestion. thanks!